packages/connector-puppeteer/src/lib/lifecycle.ts

```diff
@@ -256,7 +256,9 @@ export const close = async (browser: puppeteer.Browser, page: puppeteer.Page) =>
         return;
     }
 
-    await lock();
+    if (options && options.detached) {
```

Comment:
Wouldn't it be better to just lock when is really needed?
I think the way it is right now we will be delaying closing tabs for all of the even though it is not needed. Or is it for assurance and avoid racing conditions?

Reply:
I don' t remember exactly why the lock here, but I think it is to avoid racing conditions.